classes/PrestaShopAutoload.php

```diff
@@ -222,7 +229,7 @@ public function generateIndex()
 
         // Write classes index on disc to cache it
         $filename = static::getCacheFileIndex();
-        @mkdir(_PS_CACHE_DIR_, 0777, true);
+        @mkdir(static::getCacheDir(), 0777, true);
```

Comment:
Hmmm ... I am disturbed by this line ... Does this mean that we create a cache file that should help application to load classes and we put `777` rights on it ?

Doesn't it imply that anybody that is able to access this file may edit it and then completely take control of Prestashop ? (I mean changing autoloading behavior looks even stronger than a full db access)

Even if it is not a part of your Pull Request, I would like to ask confirmations to @PrestaShop/prestashop-core-developers because if it is what I understood, this line is bad enough to justify a security fix on all Prestashop we can ^^'

Reply:
:thinking: Maybe my thought are stupid :thinking: This line concerns only a directory ... :thinking:

Reply:
I belive 0777 is correct here, as PrestaShop may be installed via CLI (or run using php integrated webserver for example), eg. easily as a different user, anything else than 0777 will render the PrestaShop unusable.